### DISCUSSION TITLE
Are there Node Name length restrictions at 4.3.0

### DISCUSSION THREAD

**Discussion by dmoiler** [Esri/arcgis-powershell-dsc#567] [General]
Hi All,

I was performing an update to an existing 11.1 system, installing new certificates and encountered issues in which it appears node names might be trimmed / cut at 32 characters length, and so the server names can't be resolved. This is what I'm assuming based on the output below:

<img width="828" alt="image" src="https://github.com/user-attachments/assets/9fb86b7a-db85-493f-869f-4ef8840a885a">

As part of this change I have used the previous config (as indicated by @cameronkroeker) and upgraded the arcgis module to 4.3.0 (previously was using 4.2.0).

The reason for the long node names, rather than simply using the hostname is because the CN in a "workstation" certificate applies as FQDN, and so is what the WinRM needs in order for SSL to validate. If I try to use the Hostname alone it gives the warning:

<img width="565" alt="image" src="https://github.com/user-attachments/assets/e3195fbc-aa71-4a84-a456-e8a4f6dc9219">

The FQDNs (which are 34 characters in length) installed fine at 4.2.0, so wanting to know if this is a bug / known change at this version / I'm doing something wrong?

I am able to establish a CIM session in Powershell via the FQDN, so guessing it might be in the arcgis module.

Thanks!
Dean

**Comment by cameronkroeker** [comment]
Hi @dmoiler,

I want to confirm are you passing in `UseSSL` with the `Invoke-ArcGISConfiguration` command for it to use WinRM HTTPS, or is it using WinRMHTTP?

Thanks,
Cameron K.

**Reply by dmoiler** [reply]
Hi @cameronkroeker,

I'm using passing in `UseSSL` with the `Invoke-ArcGISConfiguration` command along with the `Credential` parameter.

Thanks,
Dean

**Reply by cameronkroeker** [reply]
Thanks @dmoiler for confirming this. I am not aware of any restrictions when it comes to NodeName length.

Let's try the following steps.

1. Comment out this line (save the file):

https://github.com/Esri/arcgis-powershell-dsc/blob/4e1ea910cf3c237c378dfcfc1130a296de3c4bcb/Modules/ArcGIS/ArcGIS.psm1#L259

For example:
```powershell
# if(Test-Path ".\$($ConfigurationName)") { Remove-Item ".\$($ConfigurationName)" -Force -ErrorAction Ignore -Recurse }
```

2. Launch a fresh Powershell command prompt (Right click run-as Administrator)
3. Run the `Invoke-ArcGISConfiguration` command to reproduce the error
4. In the directory where the `Invoke-ArcGISConfiguration` was ran, there should now be a `ArcGISInstall` folder containing a mof.
5. Run the following set of commands
- `$ComputeNameList = @()`
- `$ConfigurationName = "ArcGISInstall"`
- `(Get-ChildItem ".\$($ConfigurationName)\*.mof" -File -exclude @("*.meta.mof") ).Name | ForEach-Object { $ComputeNameList += $_.TrimEnd(".mof") }`
- `Get-FQDN "$ComputeNameList"`

The output should be of the entire FQDN. I ran through these steps with a 59 character FQDN and did not see any issue with characters getting trimmed.

**Reply by dmoiler** [reply]
Hi @cameronkroeker,

I've run through these steps and I think I can see what is happening.

It seems likely the trim function might be trimming the "om" from the ".com" at the end of the node name as they're characters within the ".mof" that are being trimmed.

<img width="236" alt="image" src="https://github.com/user-attachments/assets/42ec8000-92ee-4928-a5fe-4e715259a003">

<img width="1507" alt="image" src="https://github.com/user-attachments/assets/f71fe2e3-6b35-4657-b745-3d28e31e58f3">


I've tried with a few test computer names and this seems to be the outcome:

<img width="568" alt="image" src="https://github.com/user-attachments/assets/f4b12e5f-a5fd-4602-b4bf-2591efdb275b">

Hopefully this is reproduceable from you side.

Thanks,
Dean

**Reply by cameronkroeker** [reply]
Thanks @dmoiler this test/information was extremely helpful! I was also able to reproduce it as well. This is a bug that we will address in the next release of the module. In the mean time here is a workaround.

#### Change this:
https://github.com/Esri/arcgis-powershell-dsc/blob/4e1ea910cf3c237c378dfcfc1130a296de3c4bcb/Modules/ArcGIS/ArcGIS.psm1#L249

#### To this:
```powershell
(Get-ChildItem ".\$($ConfigurationName)\*.mof" -File -exclude @("*.meta.mof") ).Name | ForEach-Object { $ComputeNameList += ($_ -replace '\.mof$', '') }
```

>Note: Before re-running the Invoke-ArcGISConfiguration command be sure to launch a fresh powershell window after making the change above otherwise it may not recognize the change and use the old logic.

**Reply by dmoiler** [reply]
Hi @cameronkroeker,

Very happy to hear it! Absolutely one of the easiest bugs I've logged with Esri :slightly_smiling_face: 

I've tested this on my side with the updated code above you've supplied and that worked like a treat.

Thanks!

Dean

**Reply by cameronkroeker** [reply]
This has been addressed in DSC ArcGIS Module v4.4.0:
https://github.com/Esri/arcgis-powershell-dsc/releases/tag/v4.4.0
